tx-pool/src/process.rs

```diff
@@ -650,10 +652,50 @@ impl TxPoolService {
 
     pub(crate) async fn clear_pool(&mut self, new_snapshot: Arc<Snapshot>) {
         let mut tx_pool = self.tx_pool.write().await;
-        let config = tx_pool.config;
+        let config = Arc::clone(&tx_pool.config);
         self.last_txs_updated_at = Arc::new(AtomicU64::new(0));
         *tx_pool = TxPool::new(config, new_snapshot, Arc::clone(&self.last_txs_updated_at));
     }
+
+    pub(crate) async fn cache_pool(&self) -> Result<(), AnyError> {
+        let tx_pool = self.tx_pool.read().await;
+        let mut file = OpenOptions::new()
+            .create(true)
+            .write(true)
+            .truncate(true)
+            .open(&tx_pool.config.cache_file)
+            .map_err(|err| {
+                let errmsg = format!(
+                    "Failed to open the tx-pool cache file [{:?}]: {}",
+                    self.tx_pool_config.cache_file, err
+                );
+                OtherError::new(errmsg)
+            })?;
+        let pool_cache = {
+            let txs = TransactionVec::new_builder()
+                .extend(tx_pool.get_all_transactions().map(TransactionView::data))
+                .build();
+            TxPoolCache::new_builder()
```

Comment:
I don't see the need to introduce a new serialization structure, we can write the `Transaction` bytes to dump file line by line.

Reply:
I want to provide predictable compatibility, so may we want to cache more data or change the format of transactions.

Reply:
We can use `TransactionReader::from_compatible_slice` to resolve the format compatibility issue of transaction, and since we are using the replay strategy, I don't see why changing the structure of the tx pool itself would cause the old dump file not work?

Reply:
Maybe we will cache more data, no only transactions.

Reply:
> I don't see why changing the structure of the tx pool itself would cause the old dump file not work?

I couldn't agree with you.

IMHO, keep a version field could handle some things in future **we couldn't imagine** at the present.
**I don't know** what could be happened in future.
**If I could or someone told me, I wouldn't left them to the future, I would done them in this PR.**

p.s. In the previous commit, I saved the whole tx pool state and it works well. After discussion, I changed to save transactions only. You never know what happens next.

Reply:
We can save the version in the first line of the dump file, and dump all the transaction data in the second line by reusing `TransactionVec` which has been defined in `blockchain.mol` already, so that we can keep the compatibility of data upgrade in the future, and we don't need to define the schema file repeatedly.

Reply:
- How to define **"first line"** in a binary file?

  Maybe JSON is better than hex string, but store JSON string will introduce a lot of dependencies. 
  (JSON is my first design, it was simple but made the crate to be heavy.)

  If you insisted, I prefer to use JSON, at least it is readable; hex string has no advantages at all.

- Maybe it was not clear: there **only one (two, in fact) struct were new**, all others were imported from the `ckb-types`.

  I don't think define the schema file repeatedly is so bad.
  In fact, there are several projects copied the schema file and re-generate codes for **all** structs.
  Here, I provided an example how to use the schema file **as interfaces**, and use the rust code of these structs **without re-generate any duplicate codes**.
  I think it wasn't bad, not only for this feature, but also as a practical example..

  I had been asked several times how to add new structs with `ckb-types`.

  **If we didn't want to use it, even we had a proper scene we still tried to avoid to use it, how dare we encourage other developers to use it?** :dog: